status/flowcell.py

```diff
@@ -109,3 +114,255 @@ def get(self, flowcell_id):
+                try:
+                    fc[k] = int(d[k])
+                except ValueError:
+                    fc[k] = d[k]
+
+        if fc["TACA_run_status"] == "ongoing":
+
+            fc["experiment_name"], fc["sample_name"], fc["name"] = fc["TACA_run_path"].split("/")
+            
+            run_date, run_time, run_pos, run_fc, run_hash = fc["name"].split("_")
+
+            fc["start_date"] = datetime.strptime(str(run_date), '%Y%m%d').strftime('%Y-%m-%d')
+            fc["start_time"] = run_time #format
+            fc["position"] = run_pos
+            fc["flow_cell_id"] = run_fc
+            fc["run_id"] = run_hash #format
+
+        elif fc["TACA_run_status"] == "finished":
+
+            # Calculate new metrics
+            fc["basecalled_bases"] = fc["basecalled_pass_bases"] + fc["basecalled_fail_bases"]
+            fc["accuracy"] = str(
+                round(fc["basecalled_pass_bases"] / fc["basecalled_bases"] * 100, 2)
+                ) + " %"
+            
+            # TODO yield per pore, fetch pore count from 1st MUX scan message, LIMS or QC
+
+            # Add prefix metrics
+            metrics = [
+                "read_count",
+                "basecalled_pass_read_count",
+                "basecalled_fail_read_count",
+                "basecalled_bases",
+                "basecalled_pass_bases",
+                "basecalled_fail_bases",
+                "n50"
+            ]
+            for metric in metrics:
+                # Readable metrics
+                unit = "" if "count" in metric else "bp"
+                fc["_".join([metric, "str"])] = add_prefix(N=fc[metric], unit=unit)
+
+                # Formatted metrics
+                if "count" in metric:
+                    prefixed_unit = "M"
+                    divby = 10**6
+                elif "n50" in metric:
+                    prefixed_unit = "Kbp"
+                    divby = 10**3
+                elif "bases" in metric:
+                    prefixed_unit = "Gbp"
+                    divby = 10**9
+                else:
+                    continue
+                fc["_".join([metric, prefixed_unit])] = round(fc[metric] / divby, 2)
+
+        # Find project
+        query = re.compile("(p|P)\d{5}")
+        match = query.search(fc["experiment_name"])
+        if match:
+            fc["project"] = match.group(0).upper()
+            try:
+                fc["project_name"] = view_project[fc["project"]].rows[0].value["project_name"]
+            except:
+                # If the project ID can't fetch a project name, leave empty
+                fc["project_name"] = ""
+        else:
+            fc["project"] = ""
+            fc["project_name"] = ""
+
+        return fc
+
+
+    def fetch_barcodes(self, name):
+
+        fc_view = self.application.nanopore_runs_db.view("info/barcodes", descending=True)
+        
+        if len(fc_view[name].rows) == 1 and fc_view[name].rows[0].value:
+            d = fc_view[name].rows[0].value
```

Comment:
It would be better to choose slightly more descriptive variable names 😄

Reply:
7c15f6b1beb033c2b555da60f42c1b056e9577aa